File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.java
class Solution {
    public String solution(int[] numLog) {
        // 정답 선언 및 초기화
        String answer = "";
        // numLog의 값 변화에 따라 wasd를 입력하는 for문
        for(int i=1; i<numLog.length; i++){
            if(numLog[i] - 1 == numLog[i-1]){
                answer += "w";
            } else if(numLog[i] + 1 == numLog[i-1]){
                answer += "s";
            } else if(numLog[i] - 10 == numLog[i-1]){
                answer += "d";
            } else {
                answer += "a";
            }
        }
        
        // 정답 리턴
        return answer;
    }
}

File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.py
def solution(numLog):
    answer = ''
    i = 1
    
    while i < len(numLog):
        if numLog[i-1] - numLog[i] == -1:
            answer += 'w'
            i += 1
        elif numLog[i-1] - numLog[i] == 1:
            answer += 's'
            i += 1
        elif numLog[i-1] - numLog[i] == -10:
            answer += 'd'
            i += 1
        else:
            answer += 'a'
            i += 1
    
    return answer